File: code_examples/4-Functions/5-Default_Values/CPP/main.cpp
// Copyright (c) 2020 St. Mother Teresa HS All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program prints out your name, using default function parameters

#include <iostream>

std::string FullName(std::string firstName, std::string lastName,
                     std::string middleName = "") {
    // return the full formal name

    std::string fullName;

    fullName = firstName;
    if (middleName.size() != 0) {
        fullName = fullName + " " + middleName[0] + ".";
    }
    fullName = fullName + " " + lastName;

    return fullName;
}


int main() {
    // gets a users name and prints out their formal name

    std::string firstName;
    std::string question;
    std::string middleName = "";
    std::string lastName;
    std::string fullName;

    // input
    std::cout << "Enter your first name: ";
    std::cin >> firstName;
    std::cout << "Do you have a middle name? (y/n): ";
    std::cin >> question;
    if (question == "Y" || question == "YES" || question == "y") {
        std::cout << "Enter your middle name: ";
        std::cin >> middleName;
    }
    std::cout << "Enter your last name: ";
    std::cin >> lastName;

    // call functions
    if (middleName != "") {
        fullName = FullName(firstName, lastName, middleName);
    } else {
        fullName = FullName(firstName, lastName);
    }
    std::cout << "Your formal name is " << fullName << ".";

    std::cout << "\nDone." << std::endl;
    return 0;
}

File: code_examples/4-Functions/5-Default_Values/JavaScript/main.js
// Copyright (c) 2020 Mr. Coxall All rights reserved
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program prints out your name, using default function parameters

const prompt = require('prompt-sync')()

function FullName(firstName, lastName, middleName = "") {
    // return the full formal name
    let fullName = firstName

    if (middleName.length != 0) {
        fullName = fullName + " " + middleName[0] + "."
    }
    fullName = fullName + " " + lastName

    return fullName
}

// this function function calculates the full name
let middleName = ""

// input
let firstName = prompt('Enter your first name: ')
let question = prompt('Do you have a middle name? (y/n): ')

if (question.toUpperCase() == "Y" || question.toUpperCase() == "YES") {
    middleName = prompt('Enter your middle name: ')
}
let lastName = prompt('Enter your last name: ')

// call functions
let fullName = ""

if (middleName != "") {
    fullName = FullName(firstName, lastName, middleName)
} else {
    fullName = FullName(firstName, lastName)
}
console.log(`\nYour formal name is ${fullName}.`)

console.log("\nDone.")
